python/cudf/cudf/core/column/string.py

```diff
@@ -3987,10 +3987,21 @@ def __cuda_array_interface__(self):
     def _mimic_inplace(self, other_col, inplace=False):
         out = super()._mimic_inplace(other_col, inplace=inplace)
         return out
-    
-    def view(self, dtype):
-        return self.children[1].view(dtype)
 
+    def view(self, dtype):
+        dtype = np.dtype(dtype)
+        if (
+            self.base_children[0][self.offset]
+            * self.base_children[1].dtype.itemsize
+        ) % dtype.itemsize:
+            raise TypeError(
+                "Slice data size must divide evenly into desired dtype size"
```

Comment:
It being a slice is an implementation detail, the error message should be more generic that we got `x` bytes which isn't divisible by `y` dtype itemsize.